x/gamm/keeper/grpc_query_test.go

```diff
@@ -4,11 +4,165 @@ import (
 	gocontext "context"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	"github.com/cosmos/cosmos-sdk/types/query"
 
 	"github.com/osmosis-labs/osmosis/v12/x/gamm/types"
 )
 
+func (suite *KeeperTestSuite) TestCalcExitPoolCoinsFromShares() {
+	queryClient := suite.queryClient
+	ctx := suite.Ctx
+	poolId := suite.PrepareBalancerPool()
```

Comment:
Can we add a test with a multi-asset pool?

Reply:
PrepareBalancerPool creates a pool with 4 assets

Reply:
> > @alexanderbez @p0mvn Should whitelisting of these queries be done in this PR or do we need to create a separate PR for that? Have you guys reviewed the determinism and whatever else is needed to whitelist it?
> 
> This looks good to me to whitelist. However, let's do a separate PR to get more eyes from the rest of the team.

Excellent.  I made a new branch based off this branch and created the PR here: https://github.com/osmosis-labs/osmosis/pull/3217.  Please let me know if there is anything else i need to do to get them whitelisted.